### DISCUSSION TITLE
Before You Start

### DISCUSSION THREAD

**Discussion by Archfx** [Archfx/duo-de#52] [Announcements]
Hi everyone,

We just wanted to highlight the following points regarding this project.

1. This ROM is used daily by many people. We always try to provide the latest Android system updates.  You can read more about the security of DUO-DE system image from the blog post [here](https://archfx.me/posts/2024/12/androidsecurity/).
2. The security update date you see in the Treble app is the vendor kernel update date, which is the last kernel security patch date (we can't do anything about it since it is 4-year-old hardware).
3. This ROM contains all the play integrity checks, so you can use your banking apps without any issues.
4. If you are not familiar with the flashing commands, please do some research on them before proceeding. You can check the discussion section, issues section, and the XDA Android 14 page as well for this. 
5. **Please** be nice to people who fix issues (understand that it is not their responsibility to fix any issue, if you are nice to them, they will consider fixing them), especially lot of people from [Trebledroid](https://github.com/TrebleDroid) and individuals like [ponces](https://github.com/ponces) are working behind the screen to make it happen (You can check the [contributors](https://github.com/Archfx/duo-de/graphs/contributors) section for more).  
6. Finally, if you want to try a stable Android 14/ 15 experience on Duo devices, you have it. It has many functional and security features compared to the stock ROM and very few bugs. If you want to revert to stock 12L instructions are available [here](https://github.com/Archfx/duo-de/discussions/51).

 _**If you enjoy it, please star (⭐) the repository as a support for open-source projects like this.**_

_archfx_

**Comment by sysu3peat** [comment]
@Archfx regarding #2 in this post - this is referring to the Android Security Update being stuck on January 1, 2023?
On my stock A12L Duo 2, the Android Security Update is on September 1, 2024.
I'm trying to connect my device to access work files (MS Company Portal) - however it will only allow that if the Android Security update is on August 1, 2024 or newer.

Are we stuck on January 1, 2023 forever? And there isn't a way to have my device become work compliant?

**Reply by Archfx** [reply]
**TLDR: Unfortunately it is not possible** 

Android devices rely on two primary components to create the operating environment:

- The Linux kernel
- The Android system image

DUO-DE is only the Android system image component of this combination. Since it is based on TrebleDroid, it is compatible with many different kernel versions. Microsoft has released the kernel source code for both Surface Duo 1 and Surface Duo 2 [here](https://github.com/microsoft/surface-duo-oss). Both devices use the Linux kernel version 4.xx (this version can be confirmed by navigating to `Settings -> About Device -> Android Version -> Kernel Version`). The security patch level corresponding to the kernel is also displayed there, although this may vary unless you are on the latest Microsoft-provided update.

It is theoretically possible to update the device's kernel to newer versions (potentially 5.17+) and I also tried building it a couple of times. However, this requires significant effort from a team of developers willing to volunteer for the task. This is due to the complexity of the device tree and the fact that the Surface Duo devices are based on the Qualcomm Snapdragon 855/888 platform, which is now considered outdated.

**Comment by sir86** [comment]
its not that big thing for me but it makes my wonder too. We are on last Android but seciurity updates are 2 years old.

**Comment by dtingley11** [comment]
If you root the device, it is possible to spoof the security patch. You can use trickystore to do so.
